lib/ddtrace/profiling/pprof/string_table.rb

```diff
@@ -0,0 +1,23 @@
+require 'ddtrace/utils/sequence'
+
+module Datadog
+  module Profiling
+    module Pprof
+      # Tracks strings and returns IDs
+      class StringTable
+        def initialize
+          @sequence = Utils::Sequence.new
+          @ids = { '' => @sequence.next }
```

Comment:
Also here, for freezing this empty string:
```suggestion
          @ids = { ''.freeze => @sequence.next }
```